scripts/webhook-migration/updateWebhooks.sh

```diff
@@ -0,0 +1,182 @@
+    UUIDS[$i]=${uuid}
+  done
+}
+
+###############################################################################
+# For each UUID get an OSIO token
+# Globals:
+#   UUIDS      - User ids to iterate
+#   OSIO_TOKEN - Array which gets populated with OSIO tokens
+# Arguments:
+#   None
+# Returns:
+#   None
+###############################################################################
+getOsioTokens() {
+  for i in "${!UUIDS[@]}"; do
+    # TODO Issue #94 - How to pass the key? ATM it is a file path? Should it be read from env?
+    osio_token=$(echo "osio token -k keyfile -u ${UUIDS[$i]}")
+    echo ${osio_token}
+
+    # TODO Issue #94 - This will need to be fixed
+    # MY_SECRET is single injected OSIO token for test purposes
+    OSIO_TOKENS[$i]=$MY_SECRET
+  done
+}
+
+###############################################################################
+# For each OSIO token get the GitHub token
+# Globals:
+#   OSIO_TOKENS    - OSIO tokens to iterate
+#   GITHUB_TOKENS  - Array which gets populated with GitHub tokens
+# Arguments:
+#   None
+# Returns:
+#   None
+###############################################################################
+getGitHubTokens() {
+  for i in "${!OSIO_TOKENS[@]}"; do
+    [ -z "${OSIO_TOKENS[${i}]}"  ] && echo "WARN: Empty OSIO token for namespace ${NAME_SPACES[$i]}" >&2 && continue
+
+    gh_token=$(curl -sgSL -H "Authorization: Bearer ${OSIO_TOKENS[${i}]}" ${AUTH_URL}/api/token?for=https://github.com | jq -r '.access_token')
+    GITHUB_TOKENS[$i]=${gh_token}
+  done
+}
+
+###############################################################################
+# For each OSIO token get the GitHub token
+# See https://developer.github.com/v3/repos/hooks
+# Globals:
+#   REPOS          - List of repositories to update
+#   GITHUB_TOKENS  - For each repository its GitHub access token
+#   NAME_SPACES    - For each repository its namespace (for logging)
+# Arguments:
+#   None
+# Returns:
+#   None
+###############################################################################
+updateGitHubWebhooks() {
+  for i in "${!REPOS[@]}"; do
+    [ -z "${GITHUB_TOKENS[${i}]}"  ] && echo "WARN: Empty GitHub token for namespace ${NAME_SPACES[$i]}" >&2 && continue
+
+    hook_urls=($(curl -sgSL -H "Authorization: token ${GITHUB_TOKENS[$i]}" \
+      https://api.github.com/repos/${REPOS[$i]}/hooks | \
+      jq -r '.[] | .config.url' | tr '\n' ' '))
+
+    # That's a bit flaky since it assumes the order of hooks is always the same
+    # Seems to be this way
+    hook_ids=($(curl -sgSL -H "Authorization: token ${GITHUB_TOKENS[$i]}" \
+      https://api.github.com/repos/${REPOS[$i]}/hooks | \
+      jq -r '.[] | .id' | tr '\n' ' '))
+
+    for i in "${!hook_urls[@]}"; do
+      if [[ ${hook_urls[${i}]} =~ .*${JENKINS_OLD_URL_SUFFIX}$ ]]; then
+        response=$(curl -sgSL -H "Authorization: token ${GITHUB_TOKENS[$i]}" -X PATCH \
+          --data '{"config": {"url": "'${JENKINS_PROXY_URL}'/webhook/"}}' \
```

Comment:
I think new url is not correct - should be `https://jenkins.prod-preview.openshift.io/github-webhook/`

Reply:
+1